### DISCUSSION TITLE
Flux boundary condition on displaced mesh

### DISCUSSION THREAD

**Discussion by NateAM** [idaholab/moose#25724] [Q&A General] [answered]
I would like to implement a flux boundary condition on the displaced mesh. To do this, I was thinking I would be able to simply inherit from IntegratedBC and then implement the correct off-diagonal Jacobians (I know I could also use AD) but the displacements have not been made available. So that the integration occurs over the deformed geometry, how do I extend IntegratedBC to use the displacements? I know TensorMechanics has their pressure kernel but there seems to be a lot more going on there that (at least to me) confuses the issue.

**Comment by lindsayad** [comment]
Two things come to my mind
1. weak form integration will occur automatically on the deformed geometry as long as you specify `use_displaced_mesh = true` in your input file for your integrated boundary condition
2. When discussing implementing off-diagonal Jacobians are you referring to coding the off-diagonal Jacobians with respect to the displacements themselves? E.g. the dependence on JxW, shape function gradients, etc. on the displacement field?

**Reply by NateAM** [reply]
Hi @lindsayad 
1. Is this true even though IntegratedBC doesn't have `_displacements` defined? How can we associate the jacobian with a specific displacement dof?
2. Yes, that's what I mean. I've been successful doing this on the volume integral terms and will need to do that for the surface integral too. I know what I need to implement but without the displacement DOF ids I'm stuck. I guess I could define those degrees of freedom and see if it works.

**Reply by lindsayad** [reply, marked as answer]
> Is this true even though IntegratedBC doesn't have `_displacements` defined? 

Yes. 

> How can we associate the jacobian with a specific displacement dof?

A different question and a good one. You could either modify `IntegratedBC` to do what `KernelBase` does or you could also do it in your derived class. Either way you want to give yourself a `_displacements` member.

**Reply by NateAM** [reply]
Okay, great! I'll make those changes.

I think I'll just do it like `KernalBase` does in my own class. Thanks for the help!
